src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/ForwardFlowConflictResolver.cs

```diff
@@ -256,4 +275,65 @@ private async Task TryResolvingSourceManifestConflict(ILocalGitRepo vmr, string
         _sourceManifest.Refresh(_vmrInfo.SourceManifestPath);
         await vmr.StageAsync([_vmrInfo.SourceManifestPath], cancellationToken);
     }
+
+    private async Task ForwardFlowDependenciesAndToolset(
+        string mappingName,
+        ILocalGitRepo sourceRepo,
+        string targetBranch,
+        Codeflow lastFlow,
+        ForwardFlow currentFlow,
+        CancellationToken cancellationToken)
+    {
+        var vmr = _localGitRepoFactory.Create(_vmrInfo.VmrPath);
+
+        await _versionFileMerger.MergeJsonAsync(
+            lastFlow,
+            vmr,
+            VmrInfo.GetRelativeRepoSourcesPath(mappingName) / VersionFiles.GlobalJson,
+            lastFlow.VmrSha,
+            targetBranch,
+            sourceRepo,
+            VersionFiles.GlobalJson,
+            lastFlow.RepoSha,
+            currentFlow.RepoSha);
+
+        // and handle dotnet-tools.json if it exists
+        bool dotnetToolsConfigExists =
+            (await sourceRepo.GetFileFromGitAsync(VersionFiles.DotnetToolsConfigJson, lastFlow.RepoSha) != null) ||
+            (await vmr.GetFileFromGitAsync(VmrInfo.GetRelativeRepoSourcesPath(mappingName) / VersionFiles.DotnetToolsConfigJson, currentFlow.VmrSha) != null ||
+            (await sourceRepo.GetFileFromGitAsync(VersionFiles.DotnetToolsConfigJson, targetBranch) != null) ||
+            (await vmr.GetFileFromGitAsync(VmrInfo.GetRelativeRepoSourcesPath(mappingName) / VersionFiles.DotnetToolsConfigJson, lastFlow.VmrSha) != null));
```

Comment:
nit: Is this easier to read?

```suggestion
            (await sourceRepo.GetFileFromGitAsync(VersionFiles.DotnetToolsConfigJson, lastFlow.RepoSha) != null) ||
            (await sourceRepo.GetFileFromGitAsync(VersionFiles.DotnetToolsConfigJson, targetBranch) != null) ||
            (await vmr.GetFileFromGitAsync(VmrInfo.GetRelativeRepoSourcesPath(mappingName) / VersionFiles.DotnetToolsConfigJson, currentFlow.VmrSha) != null ||
            (await vmr.GetFileFromGitAsync(VmrInfo.GetRelativeRepoSourcesPath(mappingName) / VersionFiles.DotnetToolsConfigJson, lastFlow.VmrSha) != null));
```